### PR TITLE
Add SPI Interface for BME680

### PR DESCRIPTION
**Pull request by jraber**
Addresses issues #10 (SPI not implemented)

I copied the SPI interface from the BME280 library and only slightly modified it.

I haven't tested this, as I don't have a BME680.  It would be great is someone with access to a BME680 could give it a try.

### CODE REVIEW

**Comment by ladyada** [comment]
@brentru wanna try this? I know you had a BME680 recently!

**Comment by brentru** [comment]
@ladyada  yep! I'll test it out today on hardware

File: adafruit_bme680.py [hunk 1]
```diff
@@ -324,7 +324,7 @@ class Adafruit_BME680_I2C(Adafruit_BME680):
           will be from the previous reading."""
     def __init__(self, i2c, address=0x77, debug=False, *, refresh_rate=10):
         """Initialize the I2C device at the 'address' given"""
-        import adafruit_bus_device.i2c_device as i2c_device
+        from adafruit_bus_device import i2c_device
```
Comment (brentru) [inline adafruit_bme680.py:327]: Why are you changing the I2C import here?

Comment (ladyada) [reply adafruit_bme680.py:327]: its kinda the style we use in other drivers - so im not oppsoed to this change :D

**Comment by brentru** [comment]
Output doesn't look correct on hardware. 

code.py output:
```
Temperature: -99.5137 degrees C
Gas: 12665144 ohms
Humidity: 0%
Pressure: -91.021hPa

```
Test Setup:
> Adafruit CircuitPython 4.0.0-beta.3 on 2019-03-07; Adafruit Feather M4 Express with samd51j19

File: adafruit_bme680.py (hunk 1 above)
Comment (jraber) [reply adafruit_bme680.py:327]: Hi @brentru. I changed it mostly to make pylint happy.

When I added the SPI code, pylint complained about the SPI import, so I "fixed" it.  Not wanting the library to have two different import styles, the choice was to either change the i2c import to match the new SPI import, which would make pylint happy, or to change the SPI import to match the i2c import. Also, I feel like the imports are a bit cleaner this way.

I put it in it's own commit to make it obvious that it has nothing to do with the SPI code.

**Comment by jraber** [comment]
@brentru Thanks for testing it.  It seems that the SPI interface on the BME680 is a little bit different from the BME280.  On the BME680, we have to toggle between two memory pages, depending on the register we are accessing.

Could you retest with the latest commit?  I added the memory page toggling.  If it still gives bad values, try with debug=True.  The debug data might help figure out what exactly is going on there.

File: adafruit_bme680.py [hunk 2]
```diff
@@ -349,3 +349,56 @@ def _write(self, register, values):
...
+                buffer[2 * i + 1] = value & 0xFF
+            spi.write(buffer) #pylint: disable=no-member
+            if self._debug:
+                print("\t$%02X <= %s" % (values[0], [hex(i) for i in values[1:]]))
+
+    def _set_spi_mem_page(self, register):
+        spi_mem_page = 0x00
+        if register < 0x80:
+            spi_mem_page = 0x10
+        self._write(_BME680_REG_STATUS, spi_mem_page)
```
Comment (makermelissa) [inline adafruit_bme680.py]: Change this line to
`self._write(_BME680_REG_STATUS, bytes([spi_mem_page]))`

**Review by makermelissa — CHANGES_REQUESTED** [review]
Your change seems to fix the numbers, but the write function is expecting an array.

File: adafruit_bme680.py (hunk 2 above)
Comment (jraber) [reply adafruit_bme680.py]: I fixed it up.  Thanks testing and for the feedback.

**Review by makermelissa — APPROVED** [review]

**Comment by makermelissa** [comment]
Ok, it looks like it's working and the values were similar to the BME280.

**Pull request merged**